lib/modules/bulk_data/bulk_data_authorization_sequence.rb

```diff
@@ -0,0 +1,269 @@
+                    grant_type: 'client_credentials',
+                    client_assertion_type: 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer',
+                    iss: @instance.bulk_client_id,
+                    sub: @instance.bulk_client_id,
+                    aud: @instance.bulk_token_endpoint,
+                    exp: 5.minutes.from_now,
+                    jti: SecureRandom.hex(32))
+        id_token = JSON::JWT.new(
+          iss: iss,
+          sub: sub,
+          aud: aud,
+          exp: exp,
+          jti: jti
+        ).compact
+
+        jwk = JSON::JWK.new(JSON.parse(bulk_private_key))
+
+        id_token.header[:kid] = jwk['kid']
+        jwk_private_key = jwk.to_key
+        client_assertion = id_token.sign(jwk_private_key, 'RS384')
+
+        payload =
+          {
+            'scope' => scope,
+            'grant_type' => grant_type,
+            'client_assertion_type' => client_assertion_type,
+            'client_assertion' => client_assertion.to_s
+          }.compact
+
+        header =
+          {
+            content_type: content_type,
+            accept: 'application/json'
+          }.compact
+
+        uri = Addressable::URI.new
+        uri.query_values = payload
+
+        response = LoggedRestClient.post(@instance.bulk_token_endpoint, uri.query, header)
+        response
+      end
+
+      test :require_content_type do
+        metadata do
+          id '01'
+          name 'Bulk Data authorization request shall use content_type "application/x-www-form-urlencoded"'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/authorization/index.html#protocol-details'
+          description %(
+            After generating an authentication JWT, the client requests a new access token via HTTP POST to the FHIR authorization server’s token endpoint URL, using content-type application/x-www-form-urlencoded
+          )
+        end
+
+        response = authorize(content_type: 'application/json')
+        assert_response_bad_or_unauthorized(response)
+      end
+
+      test :require_system_scope do
+        metadata do
+          id '02'
+          name 'Bulk Data authorization request shall use "system" scope'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/authorization/index.html#scopes'
+          description %(
+            clients SHALL use “system” scopes.
+
+            System scopes have the format system/(:resourceType|*).(read|write|*)
+          )
+        end
+
+        response = authorize(scope: 'user/*.read')
+        assert_response_bad_or_unauthorized(response)
+      end
+
+      test :require_grant_type do
+        metadata do
+          id '03'
+          name 'Bulk Data authorization request shall use grand_type "client_credentials"'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/authorization/index.html#protocol-details'
+          description %(
+            * grant_type	required	Fixed value: client_credentials
```

Comment:
The extra whitespace isn't rendered in the front end. Maybe markdown table formatting would work?

Reply:
Good point. Changed to markdown table